### PR TITLE
Support CROWBAR_USER and CROWBAR_PASSWORD env

### LINKED ISSUE #29: Support CROWBAR_KEY and key config option
For staying somehow backward compatible we should add an option to provide a keyfile for a `username:password` combination, we should parse `ENV["CROWBAR_KEY"]` as well. Optionally a config file key makes sense for that as well.

### PR DESCRIPTION
**Pull request by tboerger**
This fixes #29

### CODE REVIEW

**Comment by rsalevsky** [comment]
+1 Maybe add an spec test for the `merge_keyfile` method.

**Comment by tboerger** [comment]
I think I will replace the `keyfile` option with an environment variable as the keyfile is anyway exported within `/etc/profile.d/crowbar.sh` to `CROWBAR_KEY`

File: lib/crowbar/client/config.rb [hunk 1]
```diff
@@ -63,6 +62,50 @@ def values
...
+          ENV["CROWBAR_TIMEOUT"].to_i
+        else
+          60
+        end
+      end
+
+      def default_anonymous
+        if ENV["CROWBAR_ANONYMOUS"].present?
+          [
+            true, 1, "1", "t", "T", "true", "TRUE"
```
Comment (vuntz) [inline lib/crowbar/client/config.rb]: Isn't `ENV["CROWBAR_ANONYMOUS"]` always a string?

**Comment by vuntz** [comment]
Didn't you want to keep the -c option?

File: lib/crowbar/client/config.rb (hunk 1 above)
Comment (tboerger) [inline lib/crowbar/client/config.rb]: It depends, if it's set through ruby it can be something different. That way we are always on the safe side :)

**Comment by tboerger** [comment]
> Didn't you want to keep the -c option?

I have removed this option because I thought that the default places like /etc/crowbarrc and /home/user/.crowbarrc are enough.

**Comment by vuntz** [comment]
> Didn't you want to keep the -c option?
> 
> I have removed this option because I thought that the default places like /etc/crowbarrc and /home/user/.crowbarrc are enough.

I thought your rationale for the option was to allow having multiple config files for multiple crowbar servers?

Not that I mind much, I'm just curious.

**Comment by vuntz** [comment]
+1

**Comment by tboerger** [comment]
> I thought your rationale for the option was to allow having multiple config files for multiple crowbar servers?
> 
> Not that I mind much, I'm just curious.

For that you can define multiple servers within the config. Just switch with `--alias` or `CROWBAR_ALIAS`. A config would look like that:

```
[default]
server = http://localhost:80
username = crowbar
password = crowbar

[cloud6a]
server = http://cloud6a:80
username = crowbar
password = crowbar

[cloud6b]
server = http://cloud6b:80
username = crowbar
password = crowbar
```

This can be called with `crowbarctl --alias cloud6a barclamp list` and `crowbarctl --alias cloud6b barclamp list` or with the environment variable instead of the `--alias` option.

**Comment by rsalevsky** [comment]
:+1:

**Comment by MaximilianMeister** [comment]
looks good, 2 plus ones, merging

**Pull request merged**